pyccel/codegen/printing/cwrappercode.py

```diff
@@ -265,7 +267,7 @@ def get_static_args(self, argument):
         """
 
         if self._target_language == 'fortran' and argument.rank > 0:
-            arg_address = VariableAddress(argument)
+            arg_address = ObjectAddress(argument)
```

Comment:
*cwrappercode.py line 268*

```self._target_language = 'fortran'```,  maybe this should not be changed since the issue is C specific, or is this C specific anyway? since the file name is ```cwrappercode.py```.

Reply:
The file `cwrappercode.py` takes care of printing the wrapper which links the generated code (c or Fortran) to python. If we translate the file `file.py`, the file `cwrappercode.py` prints the file `file_wrapper.c`. This file is written in C because the python interpreter is written in C. You can find documentation about C wrappers [here](https://docs.python.org/3/c-api/index.html). This file needs to call functions from the generated code. But objects are not implemented in the same way in C and Fortran. This is usually due to arrays. Fortran has arrays in the language but C doesn't so we use the `ndarrays` class instead. This means that in the wrapper we need to call the Fortran functions differently. You can find information about the fortran/c interface [here](https://stackoverflow.com/tags/fortran-iso-c-binding/info).

As this file writes C code it is pertinent to use `VariableAddress` and its replacement `ObjectAddress`.